### PR TITLE
[PPP-4108] Use of vulnerable component jackson-databind-2.8.8.jar, ja…

### PR DESCRIPTION
**Pull request by pentaho-lmartins**
…ckson-databind-2.9.2.jar, jackson-mapper-asl-1.9.2.jar CVE-2017-7525, CVE-2017-15095, CVE-2017-15095

This is one of a series of PRs. See further details in https://github.com/pentaho/maven-parent-poms/pull/66.

**Do not merge without checking the status of the JIRA case.**

### CODE REVIEW

File: plugins/file-open-save/core/pom.xml [hunk 1]
```diff
@@ -49,25 +49,10 @@
...
     <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-core</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-annotations</artifactId>
-    </dependency>
-    <dependency>
+      <!-- this is required by OSGi bean declared in src/main/resources-filtered/OSGI-INF/blueprint/blueprint.xml -->
```
Comment (pentaho-lmartins) [inline plugins/file-open-save/core/pom.xml]: @graimundo, is this OK?

File: assemblies/core/lib/pom.xml [hunk 2]
```diff
@@ -628,39 +625,6 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
```
Comment (graimundo) [inline assemblies/core/lib/pom.xml]: By removing jackson from here aren't we removing it from data-integration/lib? Doesn't this break non OSGi plugins that assume that this lib is available?
@hbfernandes ^

File: assemblies/lib/pom.xml [hunk 3]
```diff
@@ -627,39 +624,6 @@
         </exclusion>
       </exclusions>
     </dependency>
-    <dependency>
```
Comment (graimundo) [inline assemblies/lib/pom.xml]: By removing jackson from here aren't we removing it from data-integration/lib? Doesn't this break non OSGi plugins that assume that this lib is available?
@hbfernandes ^

File: plugins/repositories/core/pom.xml [hunk 4]
```diff
@@ -32,36 +31,6 @@
...
-      <version>${fasterxml.jackson.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-databind</artifactId>
-      <version>${fasterxml.jackson.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.jaxrs</groupId>
-      <artifactId>jackson-jaxrs-json-provider</artifactId>
```
Comment (graimundo) [inline plugins/repositories/core/pom.xml]: Repositories plugin has a json provider declared in its blueprint as such it seems this dependency shouldn't be removed. Please double check.
https://github.com/pentaho/pentaho-kettle/blob/master/plugins/repositories/core/src/main/resources-filtered/OSGI-INF/blueprint/beans.xml#L82

File: plugins/repositories/core/pom.xml [hunk 5]
```diff
@@ -33,34 +32,9 @@
...
-      <artifactId>jackson-annotations</artifactId>
-      <version>${fasterxml.jackson.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-databind</artifactId>
-      <version>${fasterxml.jackson.version}</version>
-    </dependency>
-    <dependency>
+      <!-- this is required by OSGi bean declared in src/main/resources-filtered/OSGI-INF/blueprint/beans.xml -->
```
Comment (pentaho-lmartins) [inline plugins/repositories/core/pom.xml]: @graimundo, is this OK?

Comment (pentaho-lmartins) [reply plugins/repositories/core/pom.xml]: `jackson-jaxrs-json-provider` will also bring `jackson-jaxrs-base` and `jackson-module-jaxb-annotations` as transitive dependencies. As far as I can see, modules from `com.fasterxml.jackson.core` are not required.

File: assemblies/core/lib/pom.xml (hunk 2 above)
Comment (pentaho-lmartins) [reply assemblies/core/lib/pom.xml]: In the last test stack build we have `jackson-annotations-2.9.5.jar`, `jackson-databind-2.9.5.jar` and `jackson-core-2.9.5.jar` in _data-integration/lib_. We have these three declared as dependencies of **kettle-engine**. Do we also want to make these as explicit dependencies of **assemblies/core/lib**? @graimundo @hbfernandes

File: assemblies/lib/pom.xml (hunk 3 above)
Comment (pentaho-lmartins) [reply assemblies/lib/pom.xml]: In the last test stack build we have `jackson-annotations-2.9.5.jar`, `jackson-databind-2.9.5.jar` and `jackson-core-2.9.5.jar` in _data-integration/lib_. We have these three declared as dependencies of **kettle-engine**. Do we also want to make these as explicit dependencies of **assemblies/lib**?
Isn't it redundant to have these as dependencies of **assemblies/lib** and **assemblies/core/lib**?
@graimundo @hbfernandes

File: assemblies/core/lib/pom.xml (hunk 2 above)
Comment (graimundo) [reply assemblies/core/lib/pom.xml]: Same comment as: https://github.com/pentaho/pentaho-platform/pull/4161#discussion_r198817326

@pentaho-lmartins I think that for the scope of this story it is ok to remove the explicit dependencies and let them come in transitively. That means we are assuming that the lib contains the necessary libs for the Pentaho code base and that we are not putting anything there just for the sake of third parties. So lets go with this solution to unblock the work and create that followup thread that we talked about. If necessary, as a result of that discussion we can re add the dependencies explicitly later on.

File: assemblies/lib/pom.xml (hunk 3 above)
Comment (graimundo) [reply assemblies/lib/pom.xml]: https://github.com/pentaho/pentaho-kettle/pull/5485#discussion_r198820920

**Pull request merged**